src/Engine/Trigger/TriggerRepeat.hpp

```diff
@@ -0,0 +1,165 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <Utility/LockBox.hpp>
+#include <vector>
+#include "../linking.hpp"
+#include "SuperTrigger.hpp"
+
+
+class Entity;
+class TriggerManager;
+
+struct EventStruct {
```

Comment:
This type currently pollutes the global namespace. Please consider placing this in a namespace.